nad_ch/use_cases.py

```diff
@@ -30,6 +30,36 @@ def list_data_providers(ctx: ApplicationContext) -> List[DataProvider]:
 
 
 def ingest_data_submission(
-        ctx: ApplicationContext, file_path: str, provider_name: str
+        ctx: ApplicationContext, file_name: str, provider_name: str
 ) -> None:
-    pass
+    if not file_name:
+        ctx.logger.error('File name required')
+        return
+
+    provider = ctx.providers.get_by_name(provider_name)
+    if not provider:
+        ctx.logger.error('Provider with that name does not exist')
+        return
+
+    ctx.storage.upload(file_name, f'{provider.name}_{file_name}')
+    url = ctx.storage.get_file_url(file_name)
+
+    submission = DataSubmission(file_name, url, provider)
+    ctx.submissions.add(submission)
+    ctx.logger.info('Submission added')
```

Comment:
I think some kind of error handling logic here - to prevent file uploads from getting stored without a corresponding submission record - would be good.